### DISCUSSION TITLE
How to update `requires_python` in `pdm.lock` when `requires-python` in `pyproject.toml` changes?

### DISCUSSION THREAD

**Discussion by pgarrison** [pdm-project/pdm#3135] [Q&A] [answered]
1. At first, I have the following in my `pyproject.toml`:
    ```
    requires-python = ">=3.9,<3.11"
    ```
    And a matching block in the `pdm.lock`:
    ```
    [[metadata.targets]]
    requires_python = ">=3.9,<3.11"
    ```
2. Then, I update the `pyproject.toml`.
    ```
    requires-python = ">=3.9,<3.12"
    ```
3. How do I tell `pdm` to re-resolve all packages to be compatible with the new range `>=3.9,<3.12`? I expected to do it with `pdm lock` or `pdm lock --update-reuse` but this kept the old range `>=3.9,<3.11`.

**Comment by pgarrison** [comment]
I had luck manually editing the `pdm.lock` file to match the new version specifier, but I hope there is a better way.

**Reply by frostming** [reply, marked as answer]
The `requires-python` in lock target is independent from project's requires-python, it only copies from the latter when being created first time. You have to run `pdm lock --update-reuse --python '>=3.9,<3.12'` to update it manually.

**Comment by ohnogaurav** [comment]
To update the Python version range in pdm.lock to match the new range >=3.9,<3.12, use the following command:
pdm lock --update
This command re-resolves all packages to be compatible with the updated range specified in pyproject.toml

**Reply by o-moe** [reply]
which pdm version are you referring to? For example, pdm 2.18.1 has no `pdm lock --update` command
